gadgets/snapshot_file/test/unit/snapshot_file_test.go

```diff
@@ -0,0 +1,183 @@
+						return
+					}
+				}
+				t.Errorf("Expected at least one event with comm == tail and pid == %d, but found none", pid)
+			},
+		},
+		// "captures_no_events_with_no_matching_filter": {
+		// 	runnerConfig: runnerConfig,
+		// 	mntnsFilterMap: func(info *utils.RunnerInfo) *ebpf.Map {
+		// 		// mnts, _ := containerutils.GetNetNs(os.Getpid())
+		// 		return utils.CreateMntNsFilterMap(t, info.MountNsID + 1)
+		// 	},
+		// 	generateEvent: generateEvent,
+		// 	validateEvent: func(t *testing.T, info *utils.RunnerInfo, pid int, events []ExpectedSnapshotFileEvent) {
+		// 		// utils.ExpectNoEvent(t, info, pid, events)
+		// 		require.Empty(t, events, "Expected no events, but got %d events", len(events))
+		// 	},
+		// },
+		// "captures_events_with_matching_filter": {
+		// 	mntnsFilterMap: func(info *utils.RunnerInfo) *ebpf.Map {
+		// 		return utils.CreateMntNsFilterMap(t, info.MountNsID)
+		// 	},
+		// 	generateEvent: generateEvent,
+		// 	validateEvent: func(t *testing.T, info *utils.RunnerInfo, pid int, events []ExpectedSnapshotFileEvent) {
+		// 		// utils.ExpectAtLeastOneEvent(func(info *utils.RunnerInfo, pid int) *ExpectedSnapshotFileEvent {
+		// 		// 	return &ExpectedSnapshotFileEvent{
+		// 		// 		Comm: "tail",
+		// 		// 		Pid:  uint32(pid),
+		// 		// 		Tid:  uint32(pid),
+		// 		// 		MntNsID: info.MountNsID,
+		// 		// 	}
+		// 		// })(t, info, pid, events)
+		// 		for _, event := range events {
+		// 			if event.Comm == "tail" && event.Pid == uint32(pid) {
+		// 				t.Logf("Found matching event: %+v\n", event)
+		// 				return
+		// 			}
+		// 		}
+		// 		t.Errorf("Expected at least one event with comm == tail and pid == %d, but found none", pid)
+
+		// 	},
+		// },
+	}
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var processId int
+
+			runner := utils.NewRunnerWithTest(t, testCase.runnerConfig)
+			var mntnsFilterMap *ebpf.Map
+			if testCase.mntnsFilterMap != nil {
+				mntnsFilterMap = testCase.mntnsFilterMap(runner.Info)
+			}
+			beforeGadgetRun := func() error {
+				utils.RunWithRunner(t, runner, func() error {
+					pid, err := testCase.generateEvent(t)
+					if err != nil {
+						return err
+					}
+					processId = pid
+					return nil
+				})
+				return nil
+			}
+			opts := gadgetrunner.GadgetRunnerOpts[ExpectedSnapshotFileEvent]{
+				Image:           "snapshot_file",
+				Timeout:         5 * time.Second,
+				MntnsFilterMap:  mntnsFilterMap,
+				BeforeGadgetRun: beforeGadgetRun,
+			}
+			gadgetRunner := gadgetrunner.NewGadgetRunner(t, opts)
+
+			gadgetRunner.RunGadget()
+			
+			t.Logf("Events received with comm: tail")
+			for _, event := range gadgetRunner.CapturedEvents {
+				if event.Comm == "tail" {
+					t.Logf("Found matching event: %+v\n", event)
+				}
+			}
+
+			testCase.validateEvent(t, runner.Info, processId, gadgetRunner.CapturedEvents)
+		})
+	}
+}
+
+func generateEvent(t *testing.T) (int, error) {
+	const filePath = "/tmp/demo.txt"
```

Comment:
With https://pkg.go.dev/testing#T.TempDir you can generate a random tempdir which will be cleaned up automatically